2024-11-29 18:10:39,262 ERROR: Exception on /login [GET] [in D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask\app.py:838]
Traceback (most recent call last):
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask\app.py", line 1473, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask\app.py", line 882, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask\app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask\app.py", line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\app\routes.py", line 88, in login
    return render_template('login.html', title='Sign In', form=form)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask\templating.py", line 150, in render_template
    return _render(app, template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask\templating.py", line 131, in _render
    rv = template.render(context)
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\jinja2\environment.py", line 1304, in render
    self.environment.handle_exception()
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\jinja2\environment.py", line 939, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "D:\PROJECTS\FlaskHabr\app\templates\login.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "D:\PROJECTS\FlaskHabr\app\templates\base.html", line 59, in top-level template code
    {% block content %}{% endblock %}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\app\templates\login.html", line 5, in block 'content'
    {{ wtf.quick_form(form) }}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\jinja2\environment.py", line 487, in getattr
    return getattr(obj, attribute)
           ^^^^^^^^^^^^^^^^^^^^^^^
jinja2.exceptions.UndefinedError: 'wtf' is undefined
2024-11-29 18:10:45,781 ERROR: Exception on /login [GET] [in D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask\app.py:838]
Traceback (most recent call last):
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask\app.py", line 1473, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask\app.py", line 882, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask\app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask\app.py", line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\app\routes.py", line 88, in login
    return render_template('login.html', title='Sign In', form=form)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask\templating.py", line 150, in render_template
    return _render(app, template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask\templating.py", line 131, in _render
    rv = template.render(context)
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\jinja2\environment.py", line 1304, in render
    self.environment.handle_exception()
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\jinja2\environment.py", line 939, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "D:\PROJECTS\FlaskHabr\app\templates\login.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "D:\PROJECTS\FlaskHabr\app\templates\base.html", line 59, in top-level template code
    {% block content %}{% endblock %}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\app\templates\login.html", line 5, in block 'content'
    {{ wtf.quick_form(form) }}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\jinja2\environment.py", line 487, in getattr
    return getattr(obj, attribute)
           ^^^^^^^^^^^^^^^^^^^^^^^
jinja2.exceptions.UndefinedError: 'wtf' is undefined
2024-11-29 18:11:11,893 INFO: Microblog startup [in D:\PROJECTS\FlaskHabr\app\__init__.py:49]
2024-11-29 18:17:34,689 INFO: Microblog startup [in D:\PROJECTS\FlaskHabr\app\__init__.py:49]
2024-11-29 18:19:17,416 INFO: Microblog startup [in D:\PROJECTS\FlaskHabr\app\__init__.py:49]
2024-12-04 13:50:31,339 INFO: Microblog startup [in D:\PROJECTS\FlaskHabr\app\__init__.py:51]
2024-12-04 13:53:05,716 INFO: Microblog startup [in D:\PROJECTS\FlaskHabr\app\__init__.py:51]
2024-12-07 21:04:03,503 INFO: Microblog startup [in D:\PROJECTS\FlaskHabr\app\__init__.py:56]
2024-12-07 21:18:20,705 INFO: Microblog startup [in D:\PROJECTS\FlaskHabr\app\__init__.py:59]
2024-12-07 21:21:33,401 INFO: Microblog startup [in D:\PROJECTS\FlaskHabr\app\__init__.py:59]
2024-12-07 21:21:34,717 ERROR: Exception on / [GET] [in D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask\app.py:838]
Traceback (most recent call last):
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask\app.py", line 1473, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask\app.py", line 882, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask\app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask\app.py", line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask_login\utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\app\routes.py", line 73, in index
    return render_template("index.html", title='Home', form=form, posts=posts.items, next_url=next_url, prev_url=prev_url)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask\templating.py", line 150, in render_template
    return _render(app, template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask\templating.py", line 131, in _render
    rv = template.render(context)
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\jinja2\environment.py", line 1304, in render
    self.environment.handle_exception()
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\jinja2\environment.py", line 939, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "D:\PROJECTS\FlaskHabr\app\templates\index.html", line 2, in top-level template code
    {% import "bootstrap_wtf.html" as wtf %}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\app\templates\base.html", line 59, in top-level template code
    {% block content %}{% endblock %}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\app\templates\index.html", line 5, in block 'content'
    <h1>{{ _('Hi, %(username)s!', username=current_user.username) }}</h1>
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\jinja2\ext.py", line 168, in _gettext_alias
    return __context.call(__context.resolve("gettext"), *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\jinja2\ext.py", line 174, in gettext
    rv = __context.call(func, __string)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask_babel\__init__.py", line 166, in <lambda>
    gettext=lambda s: get_translations().ugettext(s),
                      ^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask_babel\__init__.py", line 246, in get_translations
    return get_domain().get_translations()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask_babel\__init__.py", line 609, in get_translations
    locale = get_locale()
             ^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask_babel\__init__.py", line 263, in get_locale
    rv = babel.locale_selector()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\app\__init__.py", line 17, in get_locale
    return request.accept_language.best_match(app.config['LANGUAGES'])
           ^^^^^^^
NameError: name 'request' is not defined
2024-12-07 21:22:45,803 INFO: Microblog startup [in D:\PROJECTS\FlaskHabr\app\__init__.py:59]
